aws/cdk/lib/cdk-pipeline-stack.ts

```diff
@@ -0,0 +1,285 @@
+            ]
+          },
+          build: {
+            commands: [
+              'grunt test',
+              'sbt ${1} ${2} "project viite-main" "test:run-main fi.liikennevirasto.viite.util.DataFixture flyway_migrate"',
+              'sbt ${1} ${2} "project viite-main" "test:run-main fi.liikennevirasto.viite.util.DataFixture test"',
+              'sbt test',
+              'grunt deploy',
+              'sbt clean',
+              'sbt assembly',
+              'docker build -t $REPOSITORY_URI:latest -f aws/fargate/Dockerfile .',
+              'docker images -a',
+              'echo "Pushing the Docker image"',
+              'docker push $REPOSITORY_URI:latest',
+              'echo "Image URI: $REPOSITORY_URI:latest"',
+            ]
+          }
+        },
+        artifacts: {
+          'base-directory': 'dist',
+          files: [
+            '**/*'
+          ]
+        }
+      }),
+      vpc,
+      subnetSelection: vpcSubnets,
+      securityGroups: [securityGroup],
+      timeout: cdk.Duration.minutes(60),
+      queuedTimeout: cdk.Duration.minutes(480),
+    });
+
+    // CodeBuild Project for Deploy
+    const deployProject = new codebuild.PipelineProject(this, 'DeployProject', {
+      projectName: `viite-${environment.toLowerCase()}-deploy`,
+      environment: {
+        buildImage: codebuild.LinuxBuildImage.STANDARD_5_0,
+        privileged: true,
+      },
+      environmentVariables: {
+        ECS_CLUSTER: { value: ecsClusterName },
+        ECS_SERVICE: { value: ecsServiceName },
+      },
+      buildSpec: codebuild.BuildSpec.fromObject({
+        version: '0.2',
+        phases: {
+          build: {
+            commands: [
+              'aws ecs update-service --region eu-west-1 --cluster $ECS_CLUSTER --service $ECS_SERVICE --force-new-deployment'
+            ]
+          }
+        }
+      })
+    });
+
+    // Grant permissions to CodeBuild project
+    artifactBucket.grantReadWrite(buildProject.role!);
+    ecrRepository.grantPullPush(buildProject.role!);
+
+    // CodeArtifact permissions
+    buildProject.addToRolePolicy(new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: [
+        'codeartifact:GetAuthorizationToken',
+        'codeartifact:GetRepositoryEndpoint',
+        'codeartifact:ReadFromRepository'
+      ],
+      resources: [
+        'arn:aws:codeartifact:eu-west-1:783354560127:repository/vayla-viite/viite_npm_packages',
+        'arn:aws:codeartifact:eu-west-1:783354560127:repository/vayla-viite/viite_maven_packages',
+        'arn:aws:codeartifact:eu-west-1:783354560127:domain/vayla-viite'
+      ]
+    }));
+
+    // STS GetServiceBearerToken permission
+    buildProject.addToRolePolicy(new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: ['sts:GetServiceBearerToken'],
+      resources: ['*']
+    }));
+
+    // Add ECS permissions to Deploy project
+    deployProject.addToRolePolicy(new iam.PolicyStatement({
```

Comment:
addToRolePolicy hommat ei ole tarpeellisia jos ekana haet ecs object ja sitten käytä grant metodia.

Reply:
"Property grantDeployAction does not exist on type IBaseService" ei ilmeisesti onnistu importattuun ecs serviceen.

Reply:
Niin tää johtuu nyt siitä että sä käytät codeBuild tekemään julkaisun etkä codepipeline deploy vaihetta. 
 https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-CodeDeploy.html 

Pääset helpommalla, jos käytät valmiita vaiheita.